File: backend/app/schemas/tag.py
"""
Tag schemas - Pydantic models for Tag
"""

from pydantic import BaseModel
from datetime import datetime
from typing import Optional


class TagBase(BaseModel):
    """Base tag schema"""
    name: str
    color: str = "#808080"


class TagCreate(TagBase):
    """Schema for creating a tag"""
    pass


class TagUpdate(BaseModel):
    """Schema for updating a tag"""
    name: Optional[str] = None
    color: Optional[str] = None


class TagResponse(TagBase):
    """Schema for tag response"""
    id: int
    owner_id: int
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/lib/models/tag.dart
class Tag {
  final int id;
  final int ownerId;
  final String name;
  final String? color;
  final DateTime createdAt;

  Tag({
    required this.id,
    required this.ownerId,
    required this.name,
    this.color,
    required this.createdAt,
  });

  factory Tag.fromJson(Map<String, dynamic> json) {
    return Tag(
      id: json['id'] as int,
      ownerId: json['owner_id'] as int,
      name: json['name'] as String,
      color: json['color'] as String?,
      createdAt: DateTime.parse(json['created_at'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'owner_id': ownerId,
      'name': name,
      'color': color,
      'created_at': createdAt.toIso8601String(),
    };
  }
}
